Fix/940 bundle run time

nikhilxb opened the pull request:
- Make calculation of bundle run time consistent
- Refactor bundle state transition methods

nikhilxb committed:
Added proper metadata fields.

- Need to update calculation.

nikhilxb committed:
Refactored bundle state machine transition function

- Standardized formatting across transition functions
- Refactored `last_updated` metadata update into `update_bundle()` to
remove code duplication; added `time_codalab` calculation
- Added section dividers for better clarity.

nikhilxb committed:
Removed unused bundle state (waiting for worker startup)

- The transition method for WAITING_FOR_WORKER_STARTUP is never called
in the codebase

bkgoksel reviewed (changes requested):
Generally LGTM. Will do a second pass after the refactoring we talked about.

nikhilxb committed:
Renamed bundle transition methods

- Adopt consistent naming scheme

bkgoksel reviewed (dismissed):
LGTM - the nits, formatting, rebasing and the backward compatibility question regarding the events log

nikhilxb committed:
Moved common metadata fields to DerivedBundle

- `last_updated` and `time_codalab` are now metadata props for all
DerivedBundles
- Reformatted with Black

nikhilxb committed:
Moved metadata to NamedBundle

bkgoksel reviewed (approved):
LGTM as long as the builds pass! 

Looks like there are some unrelated files that were changed by the autoformatter black when you ran it. Ideally we wouldn't have these but looks like black recently updated some small rules and that got some stuff changed. I'm planning to freeze our version of black at what it is currently to avoid this in the future but I think it's fine if this one PR touches a couple extra files.

(BTW this is the exact reason I want to enforce black at travis-build level so the same version of black etc are guaranteed to be ran at the end of the day before any commit is merged, and we should never get these kinds of unrelated changes in PRs.)

bkgoksel committed:
Change time_updated to be int

bkgoksel committed:
Fix missing metadata in finished fn after rebase

bkgoksel committed:
Format with correct version of black

bkgoksel committed:
Set last_updated the same as created when bundle created

bkgoksel committed:
Format with correct version of black

bkgoksel committed:
Set new bundle metadata upon initialization

bkgoksel committed:
Fix bad refactor from pycharm

bkgoksel committed:
Handle case where run stats are unavailable

bkgoksel commented:
Broken up into up-to-date PRs, unneeded now.
